pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1152,6 +1170,13 @@ cdef class _Timedelta(timedelta):
         else:
             raise NotImplementedError(self._reso)
 
+    cdef bint _is_in_pytimedelta_bounds(self):
+        """
+        Check if we are within the bounds of datetime.timedelta.
+        """
+        self._ensure_components()
+        return -999999999 <= self._d and self._d <= 999999999
```

Comment:
Worth using `timedelta.max.days`?

Reply:
Also why doesn't the seconds and microseconds need to be checked?

Reply:
i hard-coded the 9s for perf.  seconds/microseconds are bounded so will always be in range